narwhals/dataframe.py

```diff
@@ -21,6 +21,8 @@
     from pathlib import Path
 
     import numpy as np
+    import pandas as pd
+    import pyarrow as pa
```

Comment:
How would this work if I am in a dev env without one of these libraries?

Reply:
IIUC it would just be reported as `Any`

Reply:
@FBruzzesi , just to give a quick demo with the latest Narwhals installed in a new venv, in which I have Polars but not Modin:
```
import narwhals as nw
from narwhals.dependencies import is_modin_dataframe, is_polars_dataframe
from narwhals.typing import IntoDataFrame

def func(df: IntoDataFrame) -> None:
    if is_modin_dataframe(df):
        reveal_type(df)
    elif is_polars_dataframe(df):
        reveal_type(df)
```
outputs
```
$ mypy t.py 
t.py:7: note: Revealed type is "Any"
t.py:9: note: Revealed type is "polars.dataframe.frame.DataFrame"
```